### PR TITLE
feat: admin can now assign users to template

### PR DESCRIPTION
**Pull request by craigzour**
# Summary | Résumé

closes https://github.com/cds-snc/platform-forms-client/issues/1203

- Added admin feature to assign users to template

Screen Shot 2022-11-04 at 2 23 33 PM

Screen Shot 2022-11-04 at 2 23 52 PM

# Test instructions | Instructions pour tester la modification

- Navigate to http://localhost:3000/admin/view-templates
- Assign users to template using the Assign button in the forms table (you may want to create fake users to see how the UI behaves)
- Using `yarn prisma:studio` you can verify that the assignations are saved in the database

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [x] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: pages/id/[form]/users.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      { action: "update", subject: "User" },
+    ]);
+
+    const formID = params?.form;
+    if (!formID || Array.isArray(formID)) return redirect(locale);
+
+    const templateWithAssociatedUsers = await getTemplateWithAssociatedUsers(ability, formID);
+    if (!templateWithAssociatedUsers) return redirect(locale);
+
+    const allUsers = await getUsers(ability);
```
Comment (bryan-robitaille) [inline pages/id/[form]/users.tsx:131]: I'd be careful about passing `allUsers` directly to the page props as it contains much more information about a user than a the page needs.
I'd suggest parsing out the properties needed by the page `{id, name}` before passing the object down as props.

File: pages/id/[form]/users.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  async ({ user: { ability }, locale, params }) => {
+    checkPrivileges(ability, [
+      { action: "update", subject: "FormRecord" },
+      { action: "update", subject: "User" },
+    ]);
+
+    const formID = params?.form;
+    if (!formID || Array.isArray(formID)) return redirect(locale);
+
+    const templateWithAssociatedUsers = await getTemplateWithAssociatedUsers(ability, formID);
```
Comment (bryan-robitaille) [inline pages/id/[form]/users.tsx:128]: The users object returned from the function contains a lot of data that isn't used.  Just a though but I wonder if the select on the users should be limited to `id` and `name`.

**Review by bryan-robitaille — CHANGES_REQUESTED** [review]

**Comment by bryan-robitaille** [comment]
Weird that my review comment didn't make it through but...

Great Work!
I would just suggest that, as a security best practice, we reduce the amount of data sent to page to only the data that will be used.

File: pages/id/[form]/users.tsx (hunk 2 above)
Comment (craigzour) [reply pages/id/[form]/users.tsx:128]: You are right we should just pass the information that we need!

File: pages/id/[form]/users.tsx (hunk 1 above)
Comment (craigzour) [reply pages/id/[form]/users.tsx:131]: Did change it :)

**Review by bryan-robitaille — APPROVED** [review]
LGTM 👍

**Pull request merged**